### PR TITLE
Update hub metrics collection resources when modified

### PR DESCRIPTION
**Pull request by coleenquadros**
https://issues.redhat.com/browse/ACM-10153

- Toggling hubselfmanagement multiple times creates addon-observability namespace in hub cluster intermittently. This PR ensure we skip any reconcile in MCO touching local-cluster object
- Update hub metrics collections resources when modified

### CODE REVIEW

File: operators/multiclusterobservability/controllers/placementrule/manifestwork.go [hunk 1]
```diff
@@ -466,114 +467,98 @@ func createUpdateResourcesForHubMetricsCollection(c client.Client, manifests []w
 		log.Info("MCO Operator is terminating, skip creating resources for hub metrics collection")
 		return nil
 	}
+
 	hubManifestCopy = make([]workv1.Manifest, len(manifests))
 	for i, manifest := range manifests {
 		obj := manifest.RawExtension.Object.DeepCopyObject()
 		hubManifestCopy[i] = workv1.Manifest{RawExtension: runtime.RawExtension{Object: obj}}
 		hubManifestCopy[i] = workv1.Manifest{RawExtension: runtime.RawExtension{Object: obj}}
 	}
```
Comment (thibaultmg) [inline operators/multiclusterobservability/controllers/placementrule/manifestwork.go:476]: This line is duplicated

File: operators/multiclusterobservability/controllers/placementrule/manifestwork.go [hunk 2]
```diff
@@ -466,114 +467,98 @@ func createUpdateResourcesForHubMetricsCollection(c client.Client, manifests []w
 		log.Info("MCO Operator is terminating, skip creating resources for hub metrics collection")
 		return nil
 	}
+
 	hubManifestCopy = make([]workv1.Manifest, len(manifests))
 	for i, manifest := range manifests {
 		obj := manifest.RawExtension.Object.DeepCopyObject()
 		hubManifestCopy[i] = workv1.Manifest{RawExtension: runtime.RawExtension{Object: obj}}
 		hubManifestCopy[i] = workv1.Manifest{RawExtension: runtime.RawExtension{Object: obj}}
```
Comment (thibaultmg) [inline operators/multiclusterobservability/controllers/placementrule/manifestwork.go:475]: Duplicated line

File: operators/multiclusterobservability/controllers/placementrule/manifestwork.go [hunk 3]
```diff
@@ -466,114 +467,98 @@ func createUpdateResourcesForHubMetricsCollection(c client.Client, manifests []w
...
+		if gvk.Kind == "ClusterRoleBinding" {
 			role := obj.(*rbacv1.ClusterRoleBinding)
-			role.Subjects[0].Namespace = config.GetDefaultNamespace()
+			if len(role.Subjects) > 0 {
+				role.Subjects[0].Namespace = config.GetDefaultNamespace()
+			}
+		}
+
+		switch obj.GetObjectKind().GroupVersionKind().Kind {
+		case "Deployment":
```
Comment (thibaultmg) [inline operators/multiclusterobservability/controllers/placementrule/manifestwork.go:502]: Why do we need these type assertions? We are not using them.

File: operators/multiclusterobservability/controllers/placementrule/manifestwork.go [hunk 4]
```diff
@@ -466,114 +467,98 @@ func createUpdateResourcesForHubMetricsCollection(c client.Client, manifests []w
...
+
+		if gvk.Kind == "ClusterRoleBinding" {
 			role := obj.(*rbacv1.ClusterRoleBinding)
-			role.Subjects[0].Namespace = config.GetDefaultNamespace()
+			if len(role.Subjects) > 0 {
+				role.Subjects[0].Namespace = config.GetDefaultNamespace()
+			}
+		}
+
+		switch obj.GetObjectKind().GroupVersionKind().Kind {
```
Comment (thibaultmg) [inline operators/multiclusterobservability/controllers/placementrule/manifestwork.go:501]: I would split the code inside the loop into two different loops. At the beginning we filter on some kinds, now on some others. The code is hard to follow.

Comment (thibaultmg) [inline operators/multiclusterobservability/controllers/placementrule/manifestwork.go:501]: I would split the code inside the loop into two different loops. At the beginning we filter on some kinds, now on some others. The code is hard to follow.

File: operators/multiclusterobservability/controllers/placementrule/manifestwork.go [hunk 5]
```diff
@@ -466,114 +467,98 @@ func createUpdateResourcesForHubMetricsCollection(c client.Client, manifests []w
...
 			return err
 		}
+
+		if k8serrors.IsNotFound(err) {
+			err = c.Create(context.TODO(), obj)
+			if err != nil {
+				log.Error(err, "Failed to create resource", "kind", obj.GetObjectKind().GroupVersionKind().Kind)
+				return err
+			}
+		} else {
```
Comment (thibaultmg) [inline operators/multiclusterobservability/controllers/placementrule/manifestwork.go:528]: This else is not needed.

File: operators/multiclusterobservability/controllers/placementrule/manifestwork.go (hunk 3 above)
Comment (coleenquadros) [reply operators/multiclusterobservability/controllers/placementrule/manifestwork.go:501]: It is used to establish the type of the object to get in line 514
